f600233d1b154d507b782f84604bcb12/update/sys_script_include_b29002b097a211106fa8b4b3f153afc4.xml

```diff
@@ -9,1186 +9,1515 @@
+                        return lineItem.idiqClin;
+                    });
+                })
+                .filter(function(lineItem, index, self) {
+                    return self.indexOf(lineItem) === index;
+                });
+
+            var uniquePeriods = [];
+            var data = uniqueClins.map(function(clin) {
+                var entry = {
+                    "CLIN Type & Classification": clin,
+                };
+                payload.templatePayload.periodsEstimate.forEach(function(
+                    estimate
+                ) {
+                    var key;
+                    if (estimate.period.periodType === "BASE") {
+                        key = "Base Period";
+                    } else if (estimate.period.periodType === "OPTION") {
+                        key = "Option " + (estimate.period.optionOrder - 1);
+                    }
+                    if (uniquePeriods.indexOf(key) === -1) {
+                        uniquePeriods.push(key);
+                    }
+                    var value = estimate.periodLineItems
+                        .filter(function(lineItem) {
+                            return lineItem.idiqClin === clin;
+                        })
+                        .reduce(function(result, lineItem) {
+                            return result + lineItem.unitPrice * lineItem.quantity;
+                        }, 0);
+                    entry[key] = value;
+                });
+
+                entry.Total = sum(entry);
+                return entry;
+            });
+
+            // Create a single object with the subtotals for each period
+            var subtotals = uniquePeriods.reduce(function(periodResult, period) {
+                periodResult[period] = data
+                    .filter(function(entry) {
+                        return entry[period];
+                    })
+                    .map(function(entry) {
+                        return entry[period];
+                    })
+                    .reduce(function(subtotalResult, subtotalValue) {
+                        return subtotalResult + subtotalValue;
+                    }, 0);
+                return periodResult;
+            }, {});
+
+            // TODO Improve to handle DITCO vs external agency and lookup specific fee %; assumes external and 1%
+            var fees = {
+                "CLIN Type & Classification": "External ordering agency fee (1%)",
+            };
+            Object.assign(
+                fees,
+                Object.keys(subtotals).reduce(function(result, key) {
+                    result[key] = subtotals[key] * 0.01;
+                    return result;
+                }, {})
+            );
+            fees.Total = sum(fees);
+            data.push(fees);
+
+            // TODO Improve to lookup specific surge %; assumes 5%
+            var surge = Object.keys(subtotals).reduce(function(result, key) {
+                result[key] = subtotals[key] * 0.05;
```

Comment:
Improvement to be made

Reply:
Fixed in https://github.com/dod-ccpo/atat-snow/pull/347/commits/95f3b3c8a7f73596ee76be9ea3ef0e2f1204f23f